### PR TITLE
APIMF-3380: add missing scaladoc and fix minor issues

### PR DESCRIPTION
**Pull request by arielmirra**

### CODE REVIEW

File: shared/src/main/scala/amf/core/client/platform/config/AMFEventListener.scala [hunk 1]
```diff
@@ -124,19 +124,28 @@ class FinishedParsingEvent(private val _internal: config.FinishedParsingEvent) e
 
 // Resolution Events
 
+/**
+  * every client invocation to the transformation logic
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/platform/config/AMFEventListener.scala]: Capitalize all.

File: shared/src/main/scala/amf/core/client/platform/config/RenderOptions.scala [hunk 2]
```diff
@@ -24,26 +24,22 @@ case class RenderOptions(private[amf] val _internal: InternalRenderOptions) {
   /** Include source maps when rendering to graph. */
   def withoutSourceMaps(): RenderOptions = _internal.withoutSourceMaps
 
+  /** Include CompactUris when rendering to graph. */
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/platform/config/RenderOptions.scala]: user doesn't know what a `CompactUri` is

File: shared/src/main/scala/amf/core/client/platform/model/document/Document.scala [hunk 3]
```diff
@@ -7,7 +7,8 @@ import amf.core.client.scala.model.document.{Document => InternalDocument}
 import scala.scalajs.js.annotation.{JSExportAll, JSExportTopLevel}
 
 /**
-  * Document model class.
+  * A [[Document]] is a parsing Unit that encodes a stand-alone DomainElement and can include references to other
+  * DomainElements that reference from the encoded DomainElement
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/platform/model/document/Document.scala]: no lo entendí mucho, pudee que cortar la oración ayude.

File: shared/src/main/scala/amf/core/client/platform/model/document/Module.scala [hunk 4]
```diff
@@ -6,9 +6,7 @@ import amf.core.client.scala.model.document.{Module => InternalModule}
 
 import scala.scalajs.js.annotation.{JSExportAll, JSExportTopLevel}
 
-/**
-  * Module model class
-  */
+/** Units containing abstract fragments that can be referenced from other fragments */
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/platform/model/document/Module.scala]: `abstract fragments` ??

File: shared/src/main/scala/amf/core/client/scala/config/AMFEventListener.scala [hunk 5]
```diff
@@ -68,16 +68,31 @@ case class ParsedModelEvent(url: String, unit: BaseUnit) extends AMFEvent with G
   override val groupKey: String = url
 }
 
+/**
+  * notifies when an [[AMFParsePlugin]] has been selected
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/scala/config/AMFEventListener.scala]: capitalize

File: shared/src/main/scala/amf/core/client/scala/config/RenderOptions.scala [hunk 6]
```diff
@@ -69,10 +69,10 @@ case class RenderOptions(
   /** Exclude AmfJsonLdSerialization when rendering to graph. */
   def withoutAmfJsonLdSerialization: RenderOptions = copy(amfJsonLdSerialization = false)
 
-  /** Include FlattenedJsonLd when rendering to graph. */
+  /** Emit regular JSON-LD serialization */
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/scala/config/RenderOptions.scala]: Not necessary, are private. Besides it should be "flattened" not regular.

File: shared/src/main/scala/amf/core/client/scala/errorhandling/IgnoringErrorHandler.scala [hunk 7]
```diff
@@ -5,13 +5,14 @@ import amf.core.internal.validation.core.ValidationSpecification
 import org.mulesoft.lexer.SourceLocation
 import org.yaml.model.{SyamlException, YError}
 
+/** Error handler that ignores constraint reports */
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/scala/errorhandling/IgnoringErrorHandler.scala]: -> `that ignors errors` is better. The user doesn't know what `contraint reports` are.

File: shared/src/main/scala/amf/core/client/scala/resource/ResourceLoader.scala [hunk 8]
```diff
@@ -6,9 +6,10 @@ import scala.concurrent.Future
 
 trait ResourceLoader {
 
-  /** Fetch specified resource and return associated content. Resource should have benn previously accepted. */
+  /** Fetch specified resource and return associated content. Resource should have been previously accepted.
+    * If the resource not exists, you should return a future failed with an ResourceNotFound exception. */
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/scala/resource/ResourceLoader.scala]: `doesn't exist, it returns a failed future due to a ResourceNotFound exception` or something like that.

Comment (tomsfernandez) [reply shared/src/main/scala/amf/core/client/scala/resource/ResourceLoader.scala]: *caused by a Resour.... could also be appropiate.

File: shared/src/main/scala/amf/core/client/scala/resource/ResourceLoader.scala [hunk 9]
```diff
@@ -6,9 +6,10 @@ import scala.concurrent.Future
...
   def fetch(resource: String): Future[Content]
 
-  /** Accepts specified resource. */
+  /** Checks if the resource loader specified resource. */
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/scala/resource/ResourceLoader.scala]: a word is missing here.

File: shared/src/main/scala/amf/core/client/scala/transform/TransformationPipelineBuilder.scala [hunk 10]
```diff
@@ -13,17 +13,13 @@ case class TransformationPipelineBuilder private (builderName: String, builderSt
...
+  /** inserts stage at the end of the pipeline */
   def append(newStage: TransformationStep): TransformationPipelineBuilder = {
     this.copy(builderSteps = builderSteps :+ newStage)
 
   }
 
-  /**
-    * inserts stage at the beginning of pipeline
-    */
+  /** inserts stage at the beginning of pipeline */
```
Comment (tomsfernandez) [inline shared/src/main/scala/amf/core/client/scala/transform/TransformationPipelineBuilder.scala]: Capitalize

**Review by tomsfernandez — CHANGES_REQUESTED** [review]

File: shared/src/main/scala/amf/core/client/platform/model/document/Module.scala (hunk 4 above)
Comment (arielmirra) [reply shared/src/main/scala/amf/core/client/platform/model/document/Module.scala]: I copied that definition from the client/scala Module class, and agree that is not very clear. How can it be better described?

**Review by tomsfernandez — APPROVED** [review]

**Pull request merged**